### PR TITLE
API 7734 Practitioner Role Create Datamart Entry

### PR DESCRIPTION
**Pull request by that-guy-is-it**
This second PR is (first non-draft one) reflects the added PractitionerRole Datamart and Entity classes, along with relevant tests.

### CODE REVIEW

File: data-query/src/test/resources/gov/va/api/health/dataquery/service/controller/practitionerrole/datamart-practitioner-role.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+  "objectType" : "PractitionerRole",
```
Comment (jblefkove-va) [inline data-query/src/test/resources/gov/va/api/health/dataquery/service/controller/practitionerrole/datamart-practitioner-role.json:2]: @nmoisan can you confirm this payload is what you expect?

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/practitionerrole/PractitionerRoleSamples.java [hunk 2]
```diff
@@ -176,6 +178,66 @@ public DatamartPractitioner practitioner(String cdwId, String locCdwId, String o
                       .build()))
           .build();
     }
+
+    public DatamartPractitionerRole practitionerRole(
+        BigInteger cdwIdNumber, char cdwIdResourceCode, String given, String family) {
```
Comment (jblefkove-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/practitionerrole/PractitionerRoleSamples.java:183]: Isn't the resource code always `P`?

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/practitionerrole/PractitionerRoleEntityTest.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.math.BigInteger;
+import org.junit.jupiter.api.Test;
+
+public class PractitionerRoleEntityTest {
+
+  @Test
+  public void practitionerRoleEntity() {
```
Comment (jblefkove-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/practitionerrole/PractitionerRoleEntityTest.java:11]: test methods don't need a visibility modifier

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/practitionerrole/DatamartPractitionerRoleTest.java [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+import gov.va.api.lighthouse.datamart.DatamartReference;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+
+public class DatamartPractitionerRoleTest {
+  private static DatamartPractitionerRole sample() {
+    return DatamartPractitionerRole.builder()
```
Comment (jblefkove-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/practitionerrole/DatamartPractitionerRoleTest.java:20]: Can this come from `PractitionerRoleSamples`?

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/practitionerrole/PractitionerRoleEntity.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  @EqualsAndHashCode.Include
+  private char cdwIdResourceCode;
+
+  @Column(name = "PractitionerIdNumber")
+  private BigInteger idNumber;
+
+  @Column(name = "PractitionerResourceCode")
+  private char resourceCode;
+
+  @Column(name = "PractitionerGivenName", nullable = true)
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/practitionerrole/PractitionerRoleEntity.java:51]: I think we can omit `nullable = true`, it's not adding much

File: data-query/src/test/resources/gov/va/api/health/dataquery/service/controller/practitionerrole/datamart-practitioner-role.json [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    "type" : "Organization",
+    "reference" : "123456:I",
+    "display" : "SOME VA MEDICAL CENTER"
+  },
+  "practitioner" : {
+    "type" : "Practitioner",
+    "reference" : "123456:S",
+    "display" : "LAST,FIRST"
+  },
+  "role" : {
```
Comment (nmoisan) [inline data-query/src/test/resources/gov/va/api/health/dataquery/service/controller/practitionerrole/datamart-practitioner-role.json:15]: `role` needs to be an array.

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/practitionerrole/PractitionerRoleSamples.java (hunk 2 above)
Comment (nmoisan) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/practitionerrole/PractitionerRoleSamples.java:183]: This is correct

Comment (that-guy-is-it) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/practitionerrole/PractitionerRoleSamples.java]: @nmoisan Hang on, I am confused, do you mean the comment @jblefkove-va made is correct, or that my code snippet above (now outdated) is correct?

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**